### PR TITLE
[mellanox] fix mlnx-sfpd initialize issue

### PR DESCRIPTION
**Pull request by keboliu**
**- What I did**
Add retry for each SDK API call during init, in some case SDK API not ready when mlnx-sfpd start to call them.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#! /usr/bin/python
```
Comment (stephenxs) [inline platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd]: not sure whether the original code should be used?

File: platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd [hunk 2]
```diff
@@ -101,34 +101,65 @@ class MlnxSfpd:
...
-
-        self.user_channel_p.type = SX_USER_CHANNEL_TYPE_FD
-        self.user_channel_p.channel.fd = self.rx_fd_p
+            if not sx_api_open_pass:
+                rc, self.handle = sx_api_open(None)
+                if rc == SX_STATUS_SUCCESS:
+                    sx_api_open_pass = True
+                    retry = 1
+                else:    
+                    log_warning("failed to open SDK API handle... retrying {}".format(retry))
```
Comment (stephenxs) [inline platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd]: suggest logging "rc" as well?

File: platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd (hunk 1 above)
Comment (keboliu) [reply platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd]: restored.

File: platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd (hunk 2 above)
Comment (keboliu) [reply platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd]: added

**Pull request closed without merging**